src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageMetadataReadmeControl.xaml.cs

```diff
@@ -0,0 +1,177 @@
+
+        private void UpdateControl(DetailedPackageMetadata oldValue, DetailedPackageMetadata newValue)
+        {
+            if (newValue is not null &&
+                (oldValue?.Id != newValue.Id ||
+                oldValue?.Version != newValue.Version))
+            {
+                var loadCts = new CancellationTokenSource();
+                var oldCts = Interlocked.Exchange(ref _loadCts, loadCts);
+                oldCts?.Cancel();
+                oldCts?.Dispose();
+
+
+                NuGetUIThreadHelper.JoinableTaskFactory
+                .RunAsync(async () =>
+                {
+                    await ReadmeViewModel.LoadReadmeAsync(newValue, _renderLocalReadme, _loadCts.Token);
+                })
+                .PostOnFailure(nameof(PackageMetadataReadmeControl));
+            }
+        }
+
+        private void PackageMetadataReadmeControl_DataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            if (e.OldValue is ReadmePreviewViewModel oldViewModel && oldViewModel is not null)
+            {
+                oldViewModel.PropertyChanged -= ViewModel_PropertyChangedAsync;
+            }
+            ReadmeViewModel.PropertyChanged += ViewModel_PropertyChangedAsync;
+        }
+
+#pragma warning disable VSTHRD100 // Avoid async void methods
+        private async void ViewModel_PropertyChangedAsync(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+#pragma warning restore VSTHRD100 // Avoid async void methods
+        {
+            var markdown = string.Empty;
+            if (ReadmeViewModel is not null)
+            {
+                if (ReadmeViewModel.CanDetermineReadmeDefined)
+                {
+                    markdown = string.IsNullOrWhiteSpace(ReadmeViewModel.ReadmeMarkdown)
+                        ? UI.Resources.Text_NoReadme
+                        : ReadmeViewModel.ReadmeMarkdown;
+                }
+            }
+
+#pragma warning disable CA1031 // Do not catch general exception types
+            try
+            {
+                await UpdateMarkdownAsync(markdown);
+            }
+            catch (Exception ex)
+            {
+                ReadmeViewModel.ErrorLoadingReadme = true;
+                descriptionMarkdownPreview.Visibility = Visibility.Collapsed;
+                await TelemetryUtility.PostFaultAsync(ex, nameof(ReadmePreviewViewModel));
+            }
+#pragma warning restore CA1031 // Do not catch general exception types
+        }
+
+        private async Task UpdateMarkdownAsync(string markDown)
+        {
+            if (_markdownPreview is null)
+            {
+#pragma warning disable CS0618 // Type or member is obsolete
+                _markdownPreview = new PreviewBuilder().Build();
+                descriptionMarkdownPreview.Content = _markdownPreview.VisualElement;
+#pragma warning restore CS0618 // Type or member is obsolete
+            }
+            if (DataContext is ReadmePreviewViewModel viewModel)
```

Comment:
We can remove one nested if by combining the conditional.

```suggestion
            if (markdown is not null && DataContext is ReadmePreviewViewModel viewModel)
```